src/kernels/scaledkernel.jl

```diff
@@ -13,11 +13,42 @@ function ScaledKernel(kernel::Tk, σ²::Tσ²=1.0) where {Tk<:Kernel,Tσ²<:Real
     return ScaledKernel{Tk, Tσ²}(kernel, [σ²])
 end
 
-kappa(k::ScaledKernel, x) = first(k.σ²) * kappa(k.kernel, x)
-
 (k::ScaledKernel)(x, y) = first(k.σ²) * k.kernel(x, y)
 
-metric(k::ScaledKernel) = metric(k.kernel)
+function kernelmatrix(κ::ScaledKernel, x::AbstractVector, y::AbstractVector)
+    return κ.σ² .* kernelmatrix(κ.kernel, x, y)
+end
+
+function kernelmatrix(κ::ScaledKernel, x::AbstractVector)
+    return κ.σ² .* kernelmatrix(κ.kernel, x)
+end
+
+function kerneldiagmatrix(κ::ScaledKernel, x::AbstractVector)
+    return κ.σ² .* kerneldiagmatrix(κ.kernel, x)
+end
+
+function kernelmatrix!(
+    K::AbstractMatrix,
+    κ::ScaledKernel,
+    x::AbstractVector,
+    y::AbstractVector,
+)
+    K = kernelmatrix!(K, κ, x, y)
```

Comment:
It feels a bit weird to me to reassign `K` here (and in the implementations below):
```suggestion
    kernelmatrix!(K, κ, x, y)
```

Reply:
It might even break if a user implements `kernelmatrix!` but does not return `K`.